### PR TITLE
[iov-bns]  Add test for transaction search

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

**Comment by ethanfrey** [comment]
Seems to have been fixed in https://github.com/tendermint/tendermint/pull/2324

We should verify the rpc works now, and then update this after updating tendermint client to 0.24 or 0.25 wherever the fix lands

**Comment by webmaster128** [comment]
Fix is included on 0.24: https://github.com/tendermint/tendermint/blob/master/CHANGELOG.md#0240

**Comment by webmaster128** [comment]
Height search works now. min/max height doe not due to a Tendermint bug. Let's merge this in to ensure one test runs and one needs to compile at least.

File: packages/iov-bns/src/bnsconnection.spec.ts [hunk 1]
```diff
@@ -325,6 +325,138 @@ describe("BnsConnection", () => {
...
+      const results = await connection.searchTx({ tags: [bnsFromOrToTag(rcptAddress)] });
+      expect(results.length).toBeGreaterThanOrEqual(1);
+      const mostRecentResult = results[results.length - 1];
+      expect(mostRecentResult.transaction.kind).toEqual(TransactionKind.Send);
+      expect((mostRecentResult.transaction as SendTx).memo).toEqual(memo);
+    }
+
+    {
+      // finds transaction using tag and height
+      const results = await connection.searchTx({ tags: [bnsFromOrToTag(rcptAddress)], height: txHeight });
```
Comment (ethanfrey) [inline packages/iov-bns/src/bnsconnection.spec.ts]: good this works at least :)

File: packages/iov-bns/src/bnsconnection.spec.ts [hunk 2]
```diff
@@ -325,6 +325,138 @@ describe("BnsConnection", () => {
...
+      expect(results.length).toBeGreaterThanOrEqual(1);
+      const mostRecentResult = results[results.length - 1];
+      expect(mostRecentResult.transaction.kind).toEqual(TransactionKind.Send);
+      expect((mostRecentResult.transaction as SendTx).memo).toEqual(memo);
+    }
+
+    connection.disconnect();
+  });
+
+  // Activate when https://github.com/tendermint/tendermint/issues/2759 is fixed
```
Comment (ethanfrey) [inline packages/iov-bns/src/bnsconnection.spec.ts]: shame about this issue

**Pull request merged**